lib/sidekiq_unique_jobs/options_with_fallback.rb

```diff
@@ -57,7 +57,7 @@ def lock_class
 
     # @return [Symbol]
     def lock_type
-      @lock_type ||= options[LOCK_KEY] || item[LOCK_KEY] || unique_type
+      @lock_type ||= item&.send(:[], LOCK_KEY) || options[LOCK_KEY] || unique_type
```

Comment:
First of all. Item can never be nil, this code would not be reached without a sidekiq Job hash. Secondly, this code doesn’t really matter. 

The item should contain the lock key from the worker. The original version made sure that in the rare case you change your worker to something else it would pick that up and use it but that might mean changing locks after the lock was created. I can see how this would potentially cause some problems. 

That said, the item can’t be anything except a hash so the send should be removed.

Reply:
@mhenrixon thanks for reviewing this. I've made a couple changes.

There was a broken spec in which item was defined as `nil` and that is why I've done it initially like this.

Since `item` should never be `nil`, I've changed that spec as well.